analysis/parser/RawTextParser/src/main/kotlin/de/maibornwolff/codecharta/parser/rawtextparser/MetricCollector.kt

```diff
@@ -58,7 +58,10 @@ class MetricCollector(
     }
 
     private fun isParsableFileExtension(path: String): Boolean {
-        return fileExtensions.isEmpty() || fileExtensions.contains(path.substringAfterLast("."))
+        return fileExtensions.isEmpty() ||
+               fileExtensions.contentEquals(arrayOf("")) ||
+               fileExtensions.contains(path.substringAfterLast(".")) ||
+               fileExtensions.contains(".${path.substringAfterLast(".")}")
```

Comment:
The two lines seem to be nearly indentical. What is the difference?

Reply:
Ok I learned it from your tests.

Reply:
Ideally, the fileExtensions Array contains "normalized" or valid file extensions.
At the moment, we always have to take care of different extensions schemas (with . and without) when we use the fileExtensions variable.

Reply:
Alternatively `fileExtensions.any{it.contains(Regex(".?${path.substringAfterLast(".")}"))}` could be used to replace both lines but I find it much more difficult to understand

Reply:
The problem with normalizing the fileExtensions array, is that we also use it to report back to the user if one of their specified file extensions have not been found in the folder. If we were to normalize it, we would need another array which serves a similar purpose to the existing one. Yes, it would be cleaner if it was normalized but I think having to deal with two separate arrays with comparable function would be worse for understandability.

Reply:
I am not sure if it would be harder to understand. But it is fine for me to keep it like it is. As soon as we duplicate this logic to check for different alternatives of the extension (with and without dots) we should refactor it.